awx/main/tasks.py

```diff
@@ -2138,8 +2141,8 @@ def build_extra_vars_file(self, project_update, private_data_dir):
             'scm_url': scm_url,
             'scm_branch': scm_branch,
             'scm_clean': project_update.scm_clean,
-            'roles_enabled': settings.AWX_ROLES_ENABLED,
-            'collections_enabled': settings.AWX_COLLECTIONS_ENABLED,
+            'roles_enabled': galaxy_creds_are_defined and settings.AWX_ROLES_ENABLED,
+            'collections_enabled': galaxy_creds_are_defined and settings.AWX_COLLECTIONS_ENABLED,
```

Comment:
I would predict users having trouble debugging this, and I would like to put a docs note for it.

@tvo318 just to throw out some initial statement and wording:

If an organization does not have any Galaxy credentials associated with it, it will not download roles or collections from requirements files. On creation of a new organization:

 - the UI will automatically associate the public Galaxy credential (for now, may change in later releases)
 - organizations created via the API need to have the public Galaxy credential added in order to make use of roles/collection requirements files.